catalog/core/catalog-core-sourcemetricsplugin/src/main/java/ddf/catalog/metrics/source/SourceMetricsImpl.java

```diff
@@ -172,414 +73,59 @@ public QueryRequest process(Source source, QueryRequest input)
-    // the
-    // URL being generated for this RRD path in the Metrics tab of Admin console.
-    rrdPath = rrdPath.replaceAll(ALPHA_NUMERIC_REGEX, "");
-    LOGGER.debug("AFTER: rrdPath = {}", rrdPath);
-
-    return rrdPath;
-  }
-
-  /**
-   * Delete the metric's MBean for the specified Source.
-   *
-   * @param sourceId
-   * @param mbeanName
-   */
-  private void deleteMetric(String sourceId, String mbeanName) {
-
-    String key = sourceId + "." + mbeanName;
-    if (metrics.containsKey(key)) {
-      metricsRegistry.remove(MetricRegistry.name(sourceId, mbeanName));
-      deleteCollector(sourceId, mbeanName);
-      metrics.remove(key);
-    } else {
-      LOGGER.debug("Did not remove metric {} because it was not in metrics map", key);
-    }
-  }
-
-  /**
-   * Delete the ddf.metrics.collector.JmxCollector for the specified Source and MBean.
-   *
-   * @param sourceId
-   * @param metricName
-   */
-  private void deleteCollector(String sourceId, String metricName) {
-    String mapKey = sourceId + "." + metricName;
-    SourceMetric sourceMetric = metrics.get(mapKey);
-    LOGGER.debug(
-        "Deleting {} ddf.metrics.collector.JmxCollector for source {}", metricName, sourceId);
-    sourceMetric.getCollector().destroy();
-    metrics.remove(mapKey);
-  }
-
-  // The types of Yammer Metrics supported
-  private enum MetricType {
-    HISTOGRAM,
-    METER
-  }
-
-  /**
-   * Inner class POJO to maintain details of each metric for each Source.
-   *
-   * @author rodgersh
-   */
-  public static class SourceMetric {
-
-    // The Yammer Metric
-    private Metric metric;
-
-    // The ddf.metrics.collector.JmxCollector polling this metric's MBean
-    private RrdJmxCollector collector;
-
-    // Whether this metric is a Histogram or Meter
-    private boolean isHistogram = false;
-
-    public SourceMetric(Metric metric, RrdJmxCollector collector) {
-      this(metric, collector, false);
-    }
-
-    public SourceMetric(Metric metric, RrdJmxCollector collector, boolean isHistogram) {
-      this.metric = metric;
-      this.collector = collector;
-      this.isHistogram = isHistogram;
-    }
-
-    public Metric getMetric() {
-      return metric;
-    }
-
-    public RrdJmxCollector getCollector() {
-      return collector;
-    }
-
-    public boolean isHistogram() {
-      return isHistogram;
-    }
+  private static void updateLatencyMetric(Map.Entry<String, Serializable> property) {
+    String key = property.getKey();
+    String source = key.substring(METRICS_SOURCE_ELAPSED_PREFIX.length());
+    DistributionSummary latency =
+        LATENCY_SUMMARIES.computeIfAbsent(
+            source,
+            src ->
+                DistributionSummary.builder(METRICS_PREFIX + "." + LATENCY_TYPE)
+                    .description("Latency of catalog source requests.")
+                    .tag(SOURCE_TAG, src)
+                    .baseUnit("milliseconds")
+                    .publishPercentiles(0.5, 0.95)
```

Comment:
I think those numbers should be stable.  I based them off the defaults shown in the Micrometer examples.
https://micrometer.io/docs/concepts#_histograms_and_percentiles

>If it’s so critical that we look at a lot of nines (and it is), why do most monitoring systems stop at the 95th or 99th percentile? The answer is simply because “it’s hard!” The data collected by most monitoring systems is usually summarized in small, five or ten second windows. This, combined with the fact that we can’t average percentiles or derive five nines from a bunch of small samples of percentiles means there’s no way to know what the 99.999th percentile for the minute or hour was. We end up throwing away a lot of good data and losing fidelity.
https://bravenewgeek.com/everything-you-know-about-latency-is-wrong/

I think Micrometer does use HdrHistogram though.